### ISSUE TITLE
Unexpected behavior with closeOnSelection, timePicker, and DateRangeInput

### ISSUE THREAD

**Issue by benrudolph** [palantir/blueprint#3376]
#### Environment

- __Package version(s)__: 
```
"@blueprintjs/core": "^3.12.0",
"@blueprintjs/datetime": "^3.7.0",
"@blueprintjs/icons": "^3.5.1",
"@blueprintjs/select": "^3.6.0",
"@blueprintjs/table": "^3.4.1",
```
- __Browser and OS versions__: 

Chrome 72.0.3626.109

#### Steps to reproduce

1. Set `closeOnSelection` to `true` for a `DateRangeInput`
1. Ensure `timePickerProps` to something so that it renders
1. Attempt to change the hours and then tab or click another time input to change

See video: https://cl.ly/3b0401fd1902

#### Actual behavior

DateRangeInput closes upon changing focus.

#### Expected behavior

Popover stays open while the user changes the times.

#### Possible solution

Understand this is a potentially difficult problem to solve since it's not necessarily clear when to close the popover if the `DateRange` has changed. The current implementation is tough UX wise for the user. One idea could be to ignore the closing of the popover while any of the time inputs have focus.

Apologies if this should be filed under FR or support, felt like it straddled all of them. In any case, thank you for the wonderful library!

**Comment by adidahiya** [comment]
I agree, the current UX is not great with `closeOnSelection={true}`. I think we need to add such an example to the docs [here](https://blueprintjs.com/docs/#datetime/daterangeinput) so that we can more easily test this scenario.

If you expect users to adjust the time in addition to date, do you really want `closeOnSelection={true}`? Not sure how you could generalize the scenario here to determine when the user has completed a selection

**Comment by adidahiya** [comment]
cc @pkwi

**Comment by dnizetic** [comment]
Any news here? This is very frustrating as it makes DateRangeInput used with hours unusable.
